RevenueCatUI/Templates/V2/ViewHelpers/Shape.swift

```diff
@@ -115,19 +115,28 @@ extension View {
                 let bottomLeft = bottomLeft,
                 let bottomRight = bottomRight,
                 topLeft > 0 || topRight > 0 || bottomLeft > 0 || bottomRight > 0 {
-                    self
-                        .applyIf(topLeft > 0) {
-                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.topLeft]))
-                        }
-                        .applyIf(topRight > 0) {
-                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.topRight]))
-                        }
-                        .applyIf(bottomLeft > 0) {
-                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.bottomLeft]))
-                        }
-                        .applyIf(bottomRight > 0) {
-                            $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.bottomRight]))
-                        }
+                if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
+                    self.clipShape(UnevenRoundedRectangle(
+                        topLeadingRadius: topLeft,
+                        bottomLeadingRadius: bottomLeft,
+                        bottomTrailingRadius: bottomRight,
+                        topTrailingRadius: topRight,
+                        style: .circular
+                    ))
+                } else {
+                    self.applyIf(topLeft > 0) {
+                        $0.clipShape(SingleRoundedCornerShape(radius: topLeft, corners: [.topLeft]))
+                    }
+                    .applyIf(topRight > 0) {
+                        $0.clipShape(SingleRoundedCornerShape(radius: topRight, corners: [.topRight]))
+                    }
+                    .applyIf(bottomLeft > 0) {
+                        $0.clipShape(SingleRoundedCornerShape(radius: bottomLeft, corners: [.bottomLeft]))
+                    }
+                    .applyIf(bottomRight > 0) {
+                        $0.clipShape(SingleRoundedCornerShape(radius: bottomRight, corners: [.bottomRight]))
```

Comment:
There was a bug here where the radius was always set to `topLeft`, fixed now.

Reply:
Oh this was so on me 🫠